src/main/java/com/ibm/disni/util/MemBuf.java

```diff
@@ -53,39 +42,10 @@ public final ByteBuffer getBuffer() {
 		return buffer;
 	}
 
-	public MemoryAllocation.MemType getType() {
-		return type;
-	}
-	
-	public String getClassname() {
-		return this.classname;
-	}	
-	
-	public void dump() {
-		dump(0, buffer.capacity());
-	}
-	
-	public void dump(int offset, int length) {
-		MemBuf.dump(buffer, offset, length);
-	}
-	
-	public static void dump(ByteBuffer buffer, int offset, int length) {
-		if (offset + length > buffer.capacity()) {
-			return;
-		}
-
-		String outputbuf = "";
-		for (int i = offset; i < offset + length; i++) {
-			byte tmp1 = buffer.get(i);
-			int tmp2 = tmp1 & 0xff;
-			outputbuf += tmp2 + " ";
-		}
-		logger.info(outputbuf);
-	}
+	public final int size() { return buffer.capacity(); }
 
 	public void free() {
-		if (memAlloc != null){
-			memAlloc.free(this);
-		}
+		MemoryAllocation.getInstance().free(this);
```

Comment:
How is free on the newly fetched singleton more robust than keeping the memory allocator in the SVC object and freeing on it (like done before). That way we guarantee that the memory is allocated and freed from the same allocated, whereas using getInstance is leaving a window for potential issues, I don't see the advantage of this change.

Reply:
Peter, you want to make those two changes so that we can merge this? The two changes being, a) implement clearing of the buffer for all SVC objects, b) revert the change with the static MemoryAllocation for the free call. Let me know.